app/services/debate_maker.rb

```diff
@@ -0,0 +1,56 @@
+class DebateMaker
+  CODE_GENERATION_MAX_RETRIES = 5
+  private_constant :CODE_GENERATION_MAX_RETRIES
+
+  def initialize(code_generator, slug_generator)
+    @code_generator = code_generator
+    @slug_generator = slug_generator
+  end
+
+  def self.call(params)
+    code_generator = CodeGenerator.new(
+      Debate::CODE_LENGTH,
+      Debate::CODE_CHARSET,
+    )
+
+    new(
+      code_generator,
+      SlugGenerator.new(:topic, code_generator)
+    ).call(params)
+  end
+
+  def call(params)
+    debate = Debate.new(params)
+    generate_code(debate)
+    generate_slug(debate)
+    debate.save
+
+    add_default_answers(debate)
+
+    debate
+  end
+
+  private
+
+  def generate_code(debate)
+    code_generator.generate(num_retries: CODE_GENERATION_MAX_RETRIES) do |code|
+      begin
+        debate.code = code
+      rescue ActiveRecord::RecordNotUnique
```

Comment:
i think that this error won't be raised even if there already exist a record with same code